site/docs/v1/tech/apis/index.adoc

```diff
@@ -117,6 +118,111 @@ Each API may document specific status codes and provide a specific reason for re
 |A lambda invocation failed during this API request. An event log will have been created with details of the exception. See [breadcrumb]#System -> Event Log#.
 |===
 
+== The PATCH HTTP Method
+
+There are three options for using `PATCH` operations. You choose between them by specifying a particular `Content-Type` on the request.
+
+.PATCH options
+[cols="1,1,1,1,1,1"]
+|===
+| Name | Content-Type | Array Behavior | Available Since | RFC Link | Client Library Support
+
+|FusionAuth default
+|`application/json`
+|Appends new values to an array. Deletion requires `GET` then `PUT`.
+|1.12.0
+|N/A
+|Yes
+
+|JSON Patch
+|`application/json-patch+json`
+|Uses operations to specify JSON modifications.
+|1.39.0
+|https://www.rfc-editor.org/rfc/rfc6902[RFC 6902]
+|No
+
+|JSON Merge Patch
+|`application/merge-patch+json`
+|If an array is provided, overwrite the existing array.
+|1.39.0
+|https://www.rfc-editor.org/rfc/rfc7396[RFC 7396]
+|No
+
+|===
+
+=== PATCH Example
+
+Here's an example of how the different options work when used to modify the roles of a link:/docs/v1/tech/apis/groups[Group] which has roles of "ceo" and "dev" to remove the "dev" role.
+
+If you are only modifying specific object fields, all three `PATCH` methods are equivalent. 
+
+==== FusionAuth Default
+
+With the default method, the correct way to remove the "dev" role is to request the group JSON, find the "ceo" role from the `roleIds` array, and use `PUT` to update the group object. 
+
+[source,shell,title=Updating an object's array with the default PATCH behavior]
+----
+API_KEY=...
+GROUP_ID=...
+APPLICATION_ID=...
+group_json=`curl -H 'Authorization: '$API_KEY \
+     -X GET \
+     https://local.fusionauth.io/api/group/$GROUP_ID`
+
+# construct new object without "dev" role
+ceoRoleId=`echo $group_json | jq '.group.roles."'$APPLICATION_ID'"[] | select(.name == "ceo")|.id'`
+name=`echo $group_json | jq '.group.name'`
+data=`echo $group_json | jq '.group.data'`
+new_group_json='{"group": {"name" : '$name', "data":'$data'}, "roleIds": ['$ceoRoleId']}'
+
+echo $new_group_json
+# PUT the modified object
+curl -H 'Authorization: '$API_KEY \
+     -X PUT \
+     -H 'Content-type: application/json' \
+     https://local.fusionauth.io/api/group/$GROUP_ID \
+     -d "$new_group_json"
+----
+
+==== JSON Patch
+
+With JSON Patch, you have a flexible set of operations that can update, remove or move fields in a JSON object. Please review https://www.rfc-editor.org/rfc/rfc6902[RFC 6902] for the full list of operations.
+
+[source,shell,title=Updating an object's array with JSON Patch]
+----
+API_KEY=...
+GROUP_ID=...
+ROLE_ID=...
+
+curl -H 'Authorization: '$API_KEY \
+     -X PATCH \
+     -H 'Content-Type: application/json-patch+json' \
+     https://local.fusionauth.io/api/group/$GROUP_ID \
+     -d '[{ "op": "add", "path": "/roleIds", "value": [ "'$ROLE_ID'" ] }]'
```

Comment:
There may be multiple ways to achieve this, but the default behavior for arrays in JSON Patch is to insert values into the array. I'm not whether this works differently because you have specified `/roleIds` as the `path` with an entire array as the `value`, but the recommended (?) way to achieve this is would be to use the `remove` operation.

```json
[
    {
        "op": "remove",
        "path": "/roles/1"
    }
]
```

Reply:
```
-d '[{ "op": "add", "path": "/roleIds", "value": [ "'$ROLE_ID'" ] }]'
```

This will actually replace `/roleIds` with this array, I think what you're trying to demonstrate is an add to the array?

To add to the array:

```
-d '[{ "op": "add", "path": "/roleIds/0", "value": "'$ROLE_ID'" }]'
```

Reply:
I'm trying to remove the `devs` role. So I could remove one role, or I could set the array value to just the `ceo` role.

I agree that removing the role is probably more intuitive. Will change it to that.